h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -598,6 +607,25 @@ public InputStream open(String path) {
     }
   }
 
+  public InputStream openSeekable(Vec vec) {
```

Comment:
The contract of the method could be improved. This operation of opening a seekable stream can fail - in many ways. One of them is the underlying implementation does not support that. This should be somehow part of the contract, so the caller is able to protect itself. E.g see how it's used in `VecParquetReader`.

The fallback does `validateHdfsConfigured` which might throw `throw new H2OIllegalArgumentException("HDFS, S3, S3N, and S3A support is not configured");`- let's add some defense lines. I wouldn't walk the way of adding `Optional`, an exception to be caught is perfectly fine.

Reply:
```patch
Index: h2o-core/src/main/java/water/persist/PersistManager.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- h2o-core/src/main/java/water/persist/PersistManager.java	(revision e9e4b94cd40cb83d8bd81a96192a7aa526a02695)
+++ h2o-core/src/main/java/water/persist/PersistManager.java	(date 1583937314435)
@@ -82,6 +82,10 @@
     }
     return false;
   }
+  
+  private boolean isHdfsConfigured(){
+    return I[Value.HDFS] == null;
+  }
 
   private void validateHdfsConfigured() {
     if (I[Value.HDFS] == null) {
@@ -607,7 +611,7 @@
     }
   }
 
-  public InputStream openSeekable(Vec vec) {
+  public InputStream openSeekable(Vec vec) throws IOException {
     if (vec instanceof FileVec) {
       FileVec fileVec = (FileVec) vec;
       final String path = fileVec.getPath();
@@ -618,11 +622,16 @@
         Persist p = I[fileVec.getBackend()];
         if (p != null && p.isSeekableOpenSupported()) {
           return p.openSeekable(path);
+        } else {
+          Log.debug("Target path not seekable. Falling back to HDFS wrapper.");
         }
       }
     }
     // fallback
-    validateHdfsConfigured();
+    if(!isHdfsConfigured()){
+      throw new IllegalArgumentException(String.format("Target path '%s' not seekable. Unable to use seekable wrapper, HDFS not configured",
+              vec._key));
+    }
     return I[Value.HDFS].wrapSeekable(vec);
   }
   
```

Reply:
```patch
Index: h2o-core/src/main/java/water/persist/PersistManager.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- h2o-core/src/main/java/water/persist/PersistManager.java	(revision e9e4b94cd40cb83d8bd81a96192a7aa526a02695)
+++ h2o-core/src/main/java/water/persist/PersistManager.java	(date 1583937559043)
@@ -82,6 +82,10 @@
     }
     return false;
   }
+  
+  private boolean isHdfsConfigured(){
+    return I[Value.HDFS] == null;
+  }
 
   private void validateHdfsConfigured() {
     if (I[Value.HDFS] == null) {
@@ -607,7 +611,14 @@
     }
   }
 
-  public InputStream openSeekable(Vec vec) {
+  /**
+   * Opens a {@link Vec} in seekable implementation of an {@link InputStream}.
+   * @param vec An instance of {@link Vec} to open
+   * @return A seekable instanceo of {@link InputStream}, never null.
+   * @throws IOException When the underlying resource does not allow seekable resource creation and all fallback solutions
+   * failed. Or on reading error.
+   */
+  public InputStream openSeekable(final Vec vec) throws IOException {
     if (vec instanceof FileVec) {
       FileVec fileVec = (FileVec) vec;
       final String path = fileVec.getPath();
@@ -618,11 +629,16 @@
         Persist p = I[fileVec.getBackend()];
         if (p != null && p.isSeekableOpenSupported()) {
           return p.openSeekable(path);
+        } else {
+          Log.debug("Target path not seekable. Falling back to HDFS wrapper.");
         }
       }
     }
     // fallback
-    validateHdfsConfigured();
+    if(!isHdfsConfigured()){
+      throw new IllegalArgumentException(String.format("Target path '%s' not seekable. Unable to use seekable wrapper, HDFS not configured",
+              vec._key));
+    }
     return I[Value.HDFS].wrapSeekable(vec);
   }
   
```